src/main/groovy/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentSelectionWithCurrent.groovy

```diff
@@ -1,13 +1,48 @@
 package com.github.benmanes.gradle.versions.updates.resolutionstrategy
 
-import groovy.transform.TupleConstructor
+import com.thoughtworks.xstream.mapper.Mapper
+import org.gradle.api.artifacts.ComponentMetadata
 import org.gradle.api.artifacts.ComponentSelection
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
 
-@TupleConstructor(includeFields=true)
-class ComponentSelectionWithCurrent {
+import javax.annotation.Nullable
 
-  final String currentVersion
+class ComponentSelectionWithCurrent implements ComponentSelection {
+
+  ComponentSelectionWithCurrent(String currentVersion, ComponentSelection delegate) {
+    this.currentVersion = currentVersion
+    this.delegate = delegate
+  }
 
-  @Delegate
```

Comment:
Just wondering, what did not work with the delegate annotation? I used it in 0.24.0 in kotlin and could access methods from `ComponentSelection` just fine.